src/unix/bsd/mod.rs

```diff
@@ -10,6 +10,11 @@ pub type sa_family_t = u8;
 pub type pthread_t = ::uintptr_t;
 pub type nfds_t = ::c_uint;
 
+#[cfg(not(target_os = "dragonfly"))]
+macro_rules! utsname_len { () => (256) }
+#[cfg(target_os = "dragonfly")]
+macro_rules! utsname_len { () => (32) }
```

Comment:
Ah well. If you're feeling ambitious you could add a case in `libc-test/build.rs` to ignore this constant, but otherwise this is fine (could you add a comment though?)